From 2b4d662a462c6b0269a6e6035ce443ec29fd860e Mon Sep 17 00:00:00 2001
From: Anna Bauza <annabauza@google.com>
Date: Tue, 26 Nov 2024 20:13:07 +0000
Subject: [PATCH] Remove flag fixAvatarCrossUserLeak

Remove flag since this is a security fix and missing the flag on security branch is causing build failure.

Bug: 341688848
Test: N/A
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:2ab1084a748a2303289624e6063d2c60d10ec922)
Merged-In: I7d92e0bdb750a5e0a81e9bfd03ea50686c82f6e0
Change-Id: I7d92e0bdb750a5e0a81e9bfd03ea50686c82f6e0
---
 .../settingslib/users/EditUserPhotoController.java     | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java b/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java
index f38e91ac0d8a..0143e5ebf116 100644
--- a/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java
+++ b/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java
@@ -134,12 +134,10 @@ public class EditUserPhotoController {
         if (Flags.avatarSync()) {
             intent.putExtra(EXTRA_IS_USER_NEW, isUserNew);
             // Fix vulnerability b/341688848 by explicitly set the class name of avatar picker.
-            if (Flags.fixAvatarCrossUserLeak()) {
-                final String packageName =
-                        mActivity.getString(R.string.config_avatar_picker_package);
-                final String className = mActivity.getString(R.string.config_avatar_picker_class);
-                intent.setClassName(packageName, className);
-            }
+            final String packageName =
+                    mActivity.getString(R.string.config_avatar_picker_package);
+            final String className = mActivity.getString(R.string.config_avatar_picker_class);
+            intent.setClassName(packageName, className);
         } else {
             // SettingsLib is used by multiple apps therefore we need to know out of all apps
             // using settingsLib which one is the one we return value to.
-- 
2.48.1.262.g85cc9f2d1e-goog

